code/game/objects/items/tools/weldingtool.dm

```diff
@@ -110,12 +110,15 @@
 	var/obj/item/bodypart/affecting = H.get_bodypart(check_zone(user.zone_selected))
 
 	if(affecting && affecting.status == BODYPART_ROBOTIC && user.a_intent != INTENT_HARM)
+		if(!H.can_inject(H, FALSE, user.zone_selected)) //If the location is covered by thick material, no healies for you.
+			to_chat(user, "<span class='warning'>[H]'s [parse_zone(user.zone_selected)] is covered!</span>")
+			return
 		if(src.use_tool(H, user, 0, volume=50, amount=1))
 			if(user == H)
 				user.visible_message("<span class='notice'>[user] starts to fix some of the dents on [H]'s [affecting.name].</span>",
 					"<span class='notice'>You start fixing some of the dents on [H == user ? "your" : "[H]'s"] [affecting.name].</span>")
```

Comment:
while at it this message seems super dumb, if user is the one healed it will display this message? Shouldn't this be the other way around?
Same in cable.dm